__tests__/helpers/url.spec.js

```diff
@@ -47,6 +47,35 @@ describe('buildFetchUrl', () => {
     expect(url).toBe('http://api.domain.com/users/123/posts');
   });
 
+  it('should handle empty optional ids by removing extra forward slashes or question marks', () => {
+    const baseUrl = 'http://api.domain.com/users/:userId?/posts/:postId';
+    const url = buildFetchUrl({ url: baseUrl, id: { someId: 'huh' } });
+    expect(url).toBe('http://api.domain.com/users/posts');
+  });
+
+  it('should handle non-empty optional ids', () => {
+    const baseUrl = 'http://api.domain.com/users/:userId?/posts/:postId';
+    const url = buildFetchUrl({ url: baseUrl, id: { userId: '123', someId: 'huh' } });
+    expect(url).toBe('http://api.domain.com/users/123/posts');
+  });
+
+  it('should handle multiple empty optional ids', () => {
+    const baseUrl = 'http://api.domain.com/users/:userId?/posts/:postId?/comments';
+    const url = buildFetchUrl({ url: baseUrl, id: { someId: 'huh' } });
+    expect(url).toBe('http://api.domain.com/users/posts/comments');
+  });
+
+  it('should add query params even when they have question marks', () => {
+    const baseUrl = 'http://api.domain.com/users/:userId';
+    const url = buildFetchUrl({
+      url: baseUrl,
+      opts: {
+        query: { userId: 'questionable?' },
+      },
+    });
+    expect(url).toBe('http://api.domain.com/users?userId=questionable?');
```

Comment:
Honestly not sure, I was mainly trying to think of all the possible cases that could cause it to fail, regardless of how realistic they are.

Reply:
Gotcha. Okay, Im thinking one more unit test for this case would be good: `http://api.domain.com/users/:user?/comments` and pass:
```
opts: {
  query: { someModifier: 'questionable?' },
},
```
without a `:user` defined and see if:
`expect(url).toBe('http://api.domain.com/users/comments?someModifier=questionable?')`

Reply:
Test fails! 
![image](https://user-images.githubusercontent.com/967811/80171605-d0b1a080-85b8-11ea-9f9d-8f91c6e55902.png)
Ampersand is being included instead of the question mark. Well at least initially that was the case. But moving the `.replace(/\?\//g, '/');` to the end of the initial `builtUrl` value (before the `addQueryParams` call) got it to pass, and all remaining tests pass as well.